### ISSUE TITLE
Quest - Deleting resources incase of edit may lead to issues

### ISSUE THREAD

**Issue by maimoonak** [opensrp/fhircore#900]
**Describe the bug**
Adding resources to [questionnaireResponse.contained](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt#L157) and [deleting resources](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt#L171) on edit might not be the right approach. It can lead to issues for cases when further data entry is also done and has reference to deleted entities. It would also lead to sync issues when some other health worker has done updates to the data and here resource has been generated with some other id.

We may also not need to [set ids etc](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt#L164) and if we can open existing questionnaire response from activity instead of this step

**Comment by ekigamba** [comment]
The reason is valid for other apps but might not be valid for the scope of Quest. Quest does not have the referenced that you put above. The expected dynamic nature of Quest questionnaires does not enables us to easily track which resource(s) changed. We have the following cases:
- Multiple resources of the same type eg. Observations. We do not have a definitive way to track which Observation between the many should be replaced with 1-of-the-edited Observations
- Resources that can be optionally generated based on the values in the Questionnaire entered eg. an edit might result in 3 observations instead of the initial 4.

Kindly review the original issue here https://github.com/opensrp/fhircore/issues/667 and suggest a better solution

**Comment by f-odhiambo** [comment]
@maimoonak Do we have a PR for this ?

**Comment by f-odhiambo** [comment]
@maimoonak is this a duplicate of issue #748 ?

**Comment by pld** [comment]
it feels close but not the same, however I think it's related enough that we will address it through #748 and we can close this, and reopen after #748 is closed if we feel it was not addressed adequately
